internal/api/annotations.go

```diff
@@ -95,3 +95,38 @@ func AbortPromotionAnnotationValue(annotations map[string]string) (*kargoapi.Abo
 	}
 	return &req, ok
 }
+
+// HasMigrationAnnotationValue checks if the AnnotationKeyMigrated annotation
+// is present in the provided annotations map and if it contains the specified
+// migration type as a key with a value of true.
+func HasMigrationAnnotationValue(annotations map[string]string, migrationType string) bool {
```

Comment:
I'm wondering it if might be more ergonomic for this to accept a `client.Object` instead of a `map[string]string`, so that the question reads more like "has this resource undergone this migration?" (And rename the func accordingly, of course.)

To complement, a similar change could be made to `AddMigrationAnnotationValue()`, with the benefit that the annotation map can be created if it doesn't already exist, which might save callers of that function some effort.